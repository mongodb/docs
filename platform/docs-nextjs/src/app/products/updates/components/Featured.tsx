'use client';
import { css, cx } from '@leafygreen-ui/emotion';
import { H3 } from '@leafygreen-ui/typography';
import { Body } from '@leafygreen-ui/typography';
import Card from '@leafygreen-ui/card';
import { palette } from '@leafygreen-ui/palette';
import { Chip } from '@leafygreen-ui/chip';
import type { ProductUpdateEntry } from '../services/contentstack';
import { useRouter } from 'next/navigation';
import { theme } from '@/styles/theme';

const featuredContainerStyle = css`
  max-width: 1440px;
  margin: 48px auto 0;
  padding: 0 24px;
  margin-bottom: 60px;

  @media ${theme.screenSize.mediumAndUp} {
    padding: 0 32px;
  }

  @media ${theme.screenSize.largeAndUp} {
    padding: 0 32px;
  }

  @media ${theme.screenSize['2XLargeAndUp']} {
    padding: 0 8px;
  }
`;

const featuredContentStyle = css`
  display: flex;
  flex-direction: row;
  gap: 16px;
  flex-wrap: wrap;
  justify-content: flex-start;
  margin-top: 26px;

  @media ${theme.screenSize.mediumAndUp} {
    gap: 16px;
  }

  @media ${theme.screenSize.largeAndUp} {
    gap: 24px;
  }
`;

const featuredCardStyle = css`
  flex: 0 0 100%;
  min-width: 0;
  padding: 24px;
  min-height: 360px;

  @media ${theme.screenSize.mediumAndUp} {
    flex: 0 0 calc(50% - 8px);
    padding: 32px;
  }

  @media ${theme.screenSize.largeAndUp} {
    flex: 0 0 calc(33.333% - 16px);
    padding: 40px;
  }
`;

const dateStyle = css`
  font-size: 13px;
  line-height: 20px;
  color: ${palette.gray.dark2};
  margin-bottom: 16px;
`;

const titleStyle = css`
  font-size: 24px;
  line-height: 32px;
  margin-bottom: 8px;
  color: ${palette.black};
`;

const bodyStyle = css`
  font-size: 16px;
  line-height: 24px;
  color: ${palette.black};
  margin-bottom: 44px;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
`;

const chipsStyle = css`
  display: flex;
  flex-direction: row;
  gap: 8px;
  flex-wrap: wrap;
`;

const chipStyle = css`
  font-size: 16px;
  padding: 2px;
`;

const Featured = ({ updates }: { updates: ProductUpdateEntry[] }) => {
  const router = useRouter();

  // Filter and limit to 3 most recent featured updates
  const limitedFeaturedUpdates = updates
    .sort((a, b) => new Date(a.created_at).getTime() - new Date(b.created_at).getTime())
    .slice(0, 3);

  return (
    <div className={cx(featuredContainerStyle)}>
      <H3>Featured Updates</H3>
      <div className={cx(featuredContentStyle)}>
        {limitedFeaturedUpdates.map((update: ProductUpdateEntry) => {
          const createdAt = update.beamer_created_at || update.created_at;
          const date = new Date(createdAt).toLocaleDateString('en-US', {
            year: 'numeric',
            month: 'long',
            day: 'numeric',
          });

          const slug = update.title
            .toLowerCase()
            .replace(/[^a-z0-9]+/g, '-')
            .replace(/(^-|-$)/g, '');

          return (
            <Card
              key={update.uid}
              className={cx(featuredCardStyle)}
              onClick={() => router.push(`/products/updates/${slug}`)}
            >
              <Body className={cx(dateStyle)}>{date}</Body>
              <H3 className={cx(titleStyle)}>{update.title}</H3>
              <Body className={cx(bodyStyle)}>{update.multi_line}</Body>
              <div className={cx(chipsStyle)}>
                {/* Display all tags from the three tag fields */}
                {[
                  ...(update.tags_category || []),
                  ...(update.tags_offerings || []),
                  ...(update.tags_product || []),
                ].map((tag: string) => (
                  <Chip key={tag} label={tag} className={cx(chipStyle)} />
                ))}
              </div>
            </Card>
          );
        })}
      </div>
    </div>
  );
};

export default Featured;
