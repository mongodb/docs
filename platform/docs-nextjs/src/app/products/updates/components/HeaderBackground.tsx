import { theme } from '@/styles/theme';
import { css, cx } from '@leafygreen-ui/emotion';
import { palette } from '@leafygreen-ui/palette';
import Image from 'next/image';

const headerBackgroundStyle = css`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: ${palette.green.light3};
  border-radius: 0 0 40px 40px;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;

  /* Tablet breakpoint: between small and medium */
  @media ${theme.screenSize.mediumAndUp} {
    height: 230px;
    border-radius: 0 0 80px 80px;
  }

  /* Desktop breakpoint: larger than medium */
  @media ${theme.screenSize.largeAndUp} {
    height: 384px;
    border-radius: 0 0 80px 80px;
  }
`;

const leftImageStyle = css`
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;

  @media ${theme.screenSize.mediumAndUp} {
    top: 10px;
  }

  @media ${theme.screenSize['2XLargeAndUp']} {
    top: 0;
  }
`;

const rightImageStyle = css`
  position: absolute;
  top: 0px;
  right: 0px;
  z-index: 1;

  @media ${theme.screenSize.mediumAndUp} {
    top: 0px;
    right: -10px;
  }

  @media ${theme.screenSize['2XLargeAndUp']} {
    top: -20px;
    right: -20px;
  }
`;

const mobileImageStyle = css`
  display: block;

  @media ${theme.screenSize.mediumAndUp} {
    display: none;
  }
`;

const tabletImageStyle = css`
  display: none;

  @media ${theme.screenSize.mediumAndUp} {
    display: block;
  }

  @media ${theme.screenSize['2XLargeAndUp']} {
    display: none;
  }
`;

const desktopImageStyle = css`
  display: none;

  @media ${theme.screenSize['2XLargeAndUp']} {
    display: block;
  }
`;

const HeaderBackground = () => {
  return (
    <div className={cx(headerBackgroundStyle)}>
      {/* Left image container */}
      <div className={cx(leftImageStyle)}>
        <Image
          src="/product-updates/header-mobile-left.svg"
          alt="header-left"
          width={220}
          height={220}
          className={cx(mobileImageStyle)}
        />
        <Image
          src="/product-updates/header-tablet-left.svg"
          alt="header-left"
          width={320}
          height={320}
          className={cx(tabletImageStyle)}
        />
        <Image
          src="/product-updates/header-desktop-left.svg"
          alt="header-left"
          width={300}
          height={300}
          className={cx(desktopImageStyle)}
        />
      </div>

      {/* Right image container */}
      <div className={cx(rightImageStyle)}>
        <Image
          src="/product-updates/header-mobile-right.svg"
          alt="header-right"
          width={160}
          height={160}
          className={cx(mobileImageStyle)}
        />
        <Image
          src="/product-updates/header-tablet-right.svg"
          alt="header-right"
          width={200}
          height={200}
          className={cx(tabletImageStyle)}
        />
        <Image
          src="/product-updates/header-desktop-right.svg"
          alt="header-right"
          width={360}
          height={360}
          className={cx(desktopImageStyle)}
        />
      </div>
    </div>
  );
};

export default HeaderBackground;
