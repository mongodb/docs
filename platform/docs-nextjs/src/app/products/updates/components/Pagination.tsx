import { css, cx } from '@leafygreen-ui/emotion';
import Icon from '@leafygreen-ui/icon';
import IconButton from '@leafygreen-ui/icon-button';
import { Option, Select } from '@leafygreen-ui/select';

interface PaginationProps {
  setCurrentPage: (page: number) => void;
  totalFilteredUpdates: number;
  currentPage: number;
}

export const Pagination = ({ totalFilteredUpdates, setCurrentPage, currentPage }: PaginationProps) => {
  const goBack = () => {
    const previousPage = currentPage - 1;
    setCurrentPage(previousPage);
  };

  const goNext = () => {
    const nextPage = currentPage + 1;
    setCurrentPage(nextPage);
  };

  return (
    <div
      className={cx(css`
        display: flex;
      `)}
    >
      <div
        className={cx(css`
          display: flex;
          align-items: center;
          gap: 8px;
        `)}
      >
        <Select
          aria-label="pagination"
          placeholder=""
          size="small"
          dropdownWidthBasis="option"
          value={String(currentPage)}
          allowDeselect={false}
          onChange={(value) => setCurrentPage(Number(value))}
        >
          {Array(totalFilteredUpdates)
            .fill(0)
            .map((_, i) => {
              const page = i + 1;
              return <Option key={`${page}`} value={`${page}`}>{`${page}`}</Option>;
            })}
        </Select>
        <span
          className={cx(css`
            font-size: 13px;
          `)}
        >
          of {totalFilteredUpdates}
        </span>
      </div>
      <div
        className={cx(css`
          display: flex;
          align-items: center;
          margin-left: 5px;
        `)}
      >
        <IconButton aria-label="previous" disabled={currentPage === 1} onClick={goBack}>
          <Icon glyph="ChevronLeft" />
        </IconButton>
        <IconButton aria-label="next" disabled={currentPage === totalFilteredUpdates} onClick={goNext}>
          <Icon glyph="ChevronRight" />
        </IconButton>
      </div>
    </div>
  );
};
