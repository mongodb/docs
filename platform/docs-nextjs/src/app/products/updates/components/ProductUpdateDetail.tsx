'use client';
import type { ProductUpdateEntry } from '../services/contentstack';
import Link from 'next/link';

export enum LinkTag {
  Docs = 'Docs',
  Blog = 'Blog',
  Tutorial = 'Tutorial',
  Press = 'Press',
  Web = 'Web',
  GitHub = 'GitHub',
}
import Icon from '@leafygreen-ui/icon';
import { palette } from '@leafygreen-ui/palette';
import { Body, H3 } from '@leafygreen-ui/typography';
import { css, cx } from '@leafygreen-ui/emotion';
import { theme } from '@/styles/theme';
import Card from '@leafygreen-ui/card';
import { useRouter } from 'next/navigation';

const containerStyle = css`
  width: 100vw;
  padding: 64px;
`;

const contentStyle = css`
  max-width: 968px;
  margin: 0 auto;
`;

const headerStyle = css`
  margin-bottom: 40px;
`;

const linkTextStyle = css`
  margin-left: 4px;
  color: ${palette.blue.base};
`;

const linkStyle = css`
  display: flex;
  align-items: center;
  flex-direction: row;
  text-decoration: none;
  font-size: 16px;
  font-weight: 500;
`;

const dateStyle = css`
  font-size: 13px;
  line-height: 20px;
  color: ${palette.gray.dark1};
  margin-bottom: 24px;

  @media ${theme.screenSize.largeAndUp} {
    font-size: 16px;
    line-height: 28px;
  }
`;

const titleStyle = css`
  font-size: 32px;
  line-height: 40px;
  margin-bottom: 12px;
  font-weight: 400;
  color: ${palette.black};

  @media ${theme.screenSize.largeAndUp} {
    font-size: 48px;
    line-height: 64px;
  }
`;

const bodyStyle = css`
  font-size: 16px;
  line-height: 28px;
  color: ${palette.black};
  margin-bottom: 40px;
`;

const relatedContentTitleStyle = css`
  font-size: 24px;
  line-height: 32px;
  margin-bottom: 24px;
  color: ${palette.black};
  font-weight: 500;
`;

const relatedContentGridStyle = css`
  display: grid;
  grid-template-columns: 1fr;
  gap: 16px;

  @media ${theme.screenSize.xLargeAndUp} {
    grid-template-columns: repeat(2, 1fr);
    gap: 24px;
  }
`;

const relatedContentCardStyle = css`
  display: flex;
  flex-direction: column;
  padding: 24px;
  cursor: pointer;
  text-decoration: none;
  transition: all 0.2s ease;
  max-width: 472px;

  &:hover {
    transform: translateY(-2px);
    box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
  }
`;

const linkLabelStyle = css`
  display: inline-flex;
  align-items: center;
  gap: 4px;
  margin-bottom: 16px;
  padding: 0 6px;
  width: fit-content;
  border-radius: 6px;
`;

const linkLabelTextStyle = css`
  font-size: 12px;
  line-height: 22px;
  font-weight: 400;
`;

const cardTitleStyle = css`
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
  color: ${palette.black};
  margin: 0;
`;

const linkLabelDocsStyle = css`
  background-color: ${palette.blue.light2};
`;

const linkLabelDocsTextStyle = css`
  color: ${palette.blue.dark2};
`;

const linkLabelBlogStyle = css`
  background-color: ${palette.green.light2};
`;

const linkLabelBlogTextStyle = css`
  color: ${palette.green.dark3};
`;

const ProductUpdateDetail = ({ update }: { update: ProductUpdateEntry }) => {
  const router = useRouter();

  const createdAt = update.beamer_created_at || update.created_at;
  const date = new Date(createdAt).toLocaleDateString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });

  const getLabel = (label: string) => {
    switch (label) {
      case 'Docs':
        return (
          <div className={cx(linkLabelStyle, linkLabelDocsStyle)}>
            <Icon glyph="Note" color={palette.blue.dark2} />
            <Body className={cx(linkLabelTextStyle, linkLabelDocsTextStyle)}>{label}</Body>
          </div>
        );
      case 'Blog':
        return (
          <div className={cx(linkLabelStyle, linkLabelBlogStyle)}>
            <Icon glyph="SMS" color={palette.green.dark2} />
            <Body className={cx(linkLabelTextStyle, linkLabelBlogTextStyle)}>{label}</Body>
          </div>
        );
      default:
        return (
          <div className={cx(linkLabelStyle, linkLabelDocsStyle)}>
            <Body className={cx(linkLabelTextStyle, linkLabelDocsTextStyle)}>{label}</Body>
          </div>
        );
    }
  };

  return (
    <div className={cx(containerStyle)}>
      <div className={cx(contentStyle)}>
        <div className={cx(headerStyle)}>
          <Link href="/products/updates" className={cx(linkStyle)}>
            <Icon glyph="ArrowLeft" className={cx(linkTextStyle)} />
            <Body className={cx(linkTextStyle)}>See all updates</Body>
          </Link>
        </div>

        <div>
          <H3 className={cx(titleStyle)}>{update.title}</H3>
          <Body className={cx(dateStyle)}>{date}</Body>
        </div>

        <Body className={cx(bodyStyle)}>{update.multi_line || 'No content available'}</Body>
        {update.link_with_label && update.link_with_label.length > 0 && (
          <>
            <H3 className={cx(relatedContentTitleStyle)}>Related Content</H3>
            <div className={cx(relatedContentGridStyle)}>
              {update.link_with_label.map((link) => {
                const label = link.label;
                return (
                  <Card
                    key={link.link.href}
                    className={cx(relatedContentCardStyle)}
                    onClick={() => router.push(link.link.href)}
                  >
                    {getLabel(label)}

                    <Body className={cx(cardTitleStyle)}>{link.link.title}</Body>
                  </Card>
                );
              })}
            </div>
          </>
        )}
      </div>
    </div>
  );
};

export default ProductUpdateDetail;
