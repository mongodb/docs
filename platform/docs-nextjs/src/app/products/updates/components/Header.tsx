'use client';
import { css, cx } from '@leafygreen-ui/emotion';
import { H2 } from '@leafygreen-ui/typography';
import { Body } from '@leafygreen-ui/typography';
// import Button from '@leafygreen-ui/button';
import { palette } from '@leafygreen-ui/palette';
import HeaderBackground from './HeaderBackground';
import { theme } from '@/styles/theme';

const headerStyle = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: 20px;
  height: 283px;
  width: 100%;
  position: relative;

  @media ${theme.screenSize.mediumAndUp} {
    height: 230px;
  }

  @media ${theme.screenSize.largeAndUp} {
    height: 384px;
  }
`;

const headerContentStyle = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 782px;
  z-index: 2;
`;

const titleStyle = css`
  font-size: 32px;
  line-height: 40px;
  color: ${palette.black};
  margin-bottom: 8px;
  max-width: 326px;
  text-align: center;

  @media ${theme.screenSize.largeAndUp} {
    font-size: 64px;
    line-height: 72px;
    max-width: 763px;
  }
`;

const bodyStyle = css`
  text-align: center;
  margin-bottom: 20px;
  font-size: 13px;
  line-height: 20px;
  color: ${palette.black};
  max-width: 326px;

  @media ${theme.screenSize.mediumAndUp} {
    max-width: 520px;
  }

  @media ${theme.screenSize.largeAndUp} {
    font-size: 16px;
    line-height: 28px;
    max-width: 763px;
  }
`;

// const buttonStyle = css`
//     font-size: 13px;
//     line-height: 20px;
//     padding: 4px 12px;

//     @media ${theme.screenSize.mediumAndUp} {
//         padding: 8px 12px;
//     }

//     @media ${theme.screenSize.largeAndUp} {
//         font-size: 18px;
//         line-height: 24px;
//         padding: 12px 16px;
//         height: 48px;
//     }
// `;

const Header = () => {
  return (
    <div className={cx(headerStyle)}>
      <HeaderBackground />
      <div className={cx(headerContentStyle)}>
        <H2 className={cx(titleStyle)}>What&apos;s New at MongoDB?</H2>
        <Body className={cx(bodyStyle)}>
          Check out the latest updates in MongoDB â€“ including improvements to the developer experience, expanded
          workload support, app modernization tools, and more.
        </Body>
        {/* <Button variant="baseGreen" className={cx(buttonStyle)}>Subscribe via RSS Feed</Button> */}
      </div>
    </div>
  );
};

export default Header;
