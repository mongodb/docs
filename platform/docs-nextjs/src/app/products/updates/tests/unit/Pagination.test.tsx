import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Pagination } from '../../components/Pagination';

const setCurrentPage = jest.fn();
const renderContext = (currentPage: number) =>
  render(<Pagination totalFilteredUpdates={2} setCurrentPage={setCurrentPage} currentPage={currentPage} />);

describe('Pagination component', () => {
  it('renders correctly', () => {
    const el = renderContext(1);
    expect(el.asFragment()).toMatchSnapshot();
  });
  it('handles pagination correctly', async () => {
    // Using userEvent over fireEvent for async capabilities
    // and to simulate a real user interaction
    const user = userEvent.setup();
    // Go to the next page (max items per page is 12)
    renderContext(1);
    const selectBtn = screen.getByTestId('leafygreen-ui-select-menubutton');
    await user.click(selectBtn);
    // Check that the previous button is disabled
    const prevBtn = screen.queryByRole('button', { name: /previous/i });
    const nextBtn = screen.queryByRole('button', { name: /next/i });
    // Since we are on the first page, the previous button should be disabled
    // and the next button should not
    expect(prevBtn).toHaveAttribute('aria-disabled', 'true');
    expect(nextBtn).toHaveAttribute('aria-disabled', 'false');
    // Let the user switch pages via the selector
    expect(await screen.findByRole('listbox')).toBeInTheDocument();
    await user.click(await screen.findByRole('option', { name: '2' }));
    expect(setCurrentPage).toHaveBeenCalledWith(2);
  });

  it('handles next prev disabled state', () => {
    renderContext(2);
    const prevBtn = screen.queryByRole('button', { name: /previous/i });
    const nextBtn = screen.queryByRole('button', { name: /next/i });
    // Since we are on the last page, the previous button should be disabled
    // and the next button should not
    expect(prevBtn).toHaveAttribute('aria-disabled', 'false');
    expect(nextBtn).toHaveAttribute('aria-disabled', 'true');
  });
});
