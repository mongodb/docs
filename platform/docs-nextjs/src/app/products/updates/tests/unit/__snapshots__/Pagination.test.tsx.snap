// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pagination component renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  padding: 0;
  margin: 0;
  background-color: transparent;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  z-index: 0;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  border-radius: 6px;
  background-color: #F9FBFA;
  border-color: #889397;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  font-weight: 500;
  height: 28px;
  font-weight: 400;
  background-color: #FFFFFF;
  width: 100%;
}

.emotion-3:focus-visible,
.emotion-3[data-focus="true"] {
  outline: none;
}

.emotion-3:active,
.emotion-3[data-active="true"],
.emotion-3:focus,
.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3:focus-visible,
.emotion-3[data-focus="true"] {
  color: #001E2B;
}

.emotion-3:hover,
.emotion-3[data-hover="true"],
.emotion-3:active,
.emotion-3[data-active="true"] {
  color: #001E2B;
  background-color: #FFFFFF;
  box-shadow: 0 0 0 3px #E8EDEB;
}

.emotion-3:focus-visible,
.emotion-3[data-focus="true"] {
  background-color: #FFFFFF;
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC;
}

.emotion-3>*:last-child {
  grid-template-columns: 1fr 16px;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-3>*:last-child>svg {
  justify-self: right;
  width: 16px;
  height: 16px;
}

.emotion-3>*:last-child>svg {
  color: #5C6C75;
}

.emotion-3>*:last-child {
  padding: 0 4px 0 8px;
}

.emotion-3:focus-visible {
  box-shadow: 0 0 0 3px #0498EC;
  border-color: rgba(255, 255, 255, 0);
}

@media only screen and (max-width: 768px) and (hover: none) and (pointer: coarse),@media only screen and (max-width: 768px) and (hover: none) and (pointer: none) {
  .emotion-3 {
    height: 36px;
    font-size: 16px;
  }
}

.emotion-4 {
  overflow: hidden;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 5px;
}

.emotion-5 {
  display: grid;
  grid-auto-flow: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 0;
  -webkit-transition: all 150 ease-in-out;
  transition: all 150 ease-in-out;
  padding: 0 11px;
  gap: 6px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  gap: 4px;
  overflow: hidden;
}

.emotion-7 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 100%;
}

.emotion-8 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-9 {
  color: #889397;
  height: 16px;
  width: 16px;
  justify-self: left;
}

.emotion-10 {
  display: none;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
}

.emotion-12 {
  font-size: 13px;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 5px;
}

.emotion-14 {
  border: none;
  -webkit-appearance: unset;
  padding: unset;
  display: inline-block;
  border-radius: 100px;
  position: relative;
  cursor: pointer;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: 150ms ease-in-out;
  transition: 150ms ease-in-out;
  transition-property: color,box-shadow;
  background-color: rgba(255, 255, 255, 0);
  height: 28px;
  width: 28px;
  color: #5C6C75;
  cursor: not-allowed;
  color: #C1C7C6;
  background-color: rgba(255, 255, 255, 0);
}

.emotion-14::before {
  content: '';
  -webkit-transition: 150ms all ease-in-out;
  transition: 150ms all ease-in-out;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: 100%;
  -webkit-transform: scale(0.8);
  -moz-transform: scale(0.8);
  -ms-transform: scale(0.8);
  transform: scale(0.8);
}

.emotion-14:active:before,
.emotion-14:hover:before,
.emotion-14:focus:before,
.emotion-14[data-hover='true']:before,
.emotion-14[data-focus='true']:before {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-14:focus {
  outline: none;
}

.emotion-14:active,
.emotion-14:hover,
.emotion-14[data-hover='true'] {
  color: #001E2B;
}

.emotion-14:active::before,
.emotion-14:hover::before,
.emotion-14[data-hover='true']::before {
  background-color: rgba(61,79,88,0.1);
}

.emotion-14:focus-visible,
.emotion-14[data-focus='true'] {
  color: #001E2B;
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC;
}

.emotion-14:focus-visible::before,
.emotion-14[data-focus='true']::before {
  background-color: rgba(61,79,88,0.1);
}

.emotion-14:active,
.emotion-14:hover,
.emotion-14[data-hover='true'] {
  color: #C1C7C6;
}

.emotion-14:active::before,
.emotion-14:hover::before,
.emotion-14[data-hover='true']::before {
  background-color: rgba(255, 255, 255, 0);
}

.emotion-14:focus,
.emotion-14[data-focus='true'] {
  color: #C1C7C6;
}

.emotion-14:focus::before,
.emotion-14[data-focus='true']::before {
  background-color: rgba(255, 255, 255, 0);
}

.emotion-15 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-16 {
  border: none;
  -webkit-appearance: unset;
  padding: unset;
  display: inline-block;
  border-radius: 100px;
  position: relative;
  cursor: pointer;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: 150ms ease-in-out;
  transition: 150ms ease-in-out;
  transition-property: color,box-shadow;
  background-color: rgba(255, 255, 255, 0);
  height: 28px;
  width: 28px;
  color: #5C6C75;
}

.emotion-16::before {
  content: '';
  -webkit-transition: 150ms all ease-in-out;
  transition: 150ms all ease-in-out;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: 100%;
  -webkit-transform: scale(0.8);
  -moz-transform: scale(0.8);
  -ms-transform: scale(0.8);
  transform: scale(0.8);
}

.emotion-16:active:before,
.emotion-16:hover:before,
.emotion-16:focus:before,
.emotion-16[data-hover='true']:before,
.emotion-16[data-focus='true']:before {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-16:focus {
  outline: none;
}

.emotion-16:active,
.emotion-16:hover,
.emotion-16[data-hover='true'] {
  color: #001E2B;
}

.emotion-16:active::before,
.emotion-16:hover::before,
.emotion-16[data-hover='true']::before {
  background-color: rgba(61,79,88,0.1);
}

.emotion-16:focus-visible,
.emotion-16[data-focus='true'] {
  color: #001E2B;
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC;
}

.emotion-16:focus-visible::before,
.emotion-16[data-focus='true']::before {
  background-color: rgba(61,79,88,0.1);
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
        data-lgid="lg-select"
        data-testid="lg-select"
      >
        <button
          aria-controls="select-1-menu"
          aria-describedby="select-1-description"
          aria-disabled="false"
          aria-expanded="false"
          aria-invalid="false"
          aria-label="pagination"
          class="lg-ui-button-0000 emotion-3"
          data-lgid="lg-select-trigger"
          data-testid="leafygreen-ui-select-menubutton"
          id="select-2"
          type="button"
          value="1"
        >
          <div
            class="emotion-4"
          />
          <div
            class="emotion-5"
          >
            <div
              class="emotion-6"
            >
              <div
                class="lg-ui-select-menu-0000 emotion-7"
                data-lgid="lg-select-button_text"
                data-testid="lg-select-button_text"
              >
                1
              </div>
            </div>
            <svg
              alt=""
              aria-hidden="true"
              class="emotion-8 emotion-9"
              height="16"
              role="presentation"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M8.67903 10.7962C8.45271 11.0679 8.04729 11.0679 7.82097 10.7962L4.63962 6.97649C4.3213 6.59428 4.5824 6 5.06866 6L11.4313 6C11.9176 6 12.1787 6.59428 11.8604 6.97649L8.67903 10.7962Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </button>
        <span
          class="emotion-10"
        />
        <div
          aria-live="polite"
          aria-relevant="all"
          class="emotion-11"
          data-lgid="lg-form_field-feedback"
          data-testid="lg-form_field-feedback"
        />
      </div>
      <span
        class="emotion-12"
      >
        of 2
      </span>
    </div>
    <div
      class="emotion-13"
    >
      <button
        aria-disabled="true"
        aria-label="previous"
        class="emotion-14"
        tabindex="0"
      >
        <div
          class="emotion-15"
        >
          <svg
            aria-label="Chevron Left Icon"
            class=""
            fill="none"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10.778 1.636a1 1 0 0 0-1.414 0l-4.95 4.95-.707.707a1 1 0 0 0 0 1.414l.707.707 4.95 4.95a1 1 0 0 0 1.414 0l.707-.707a1 1 0 0 0 0-1.414L7.243 8l4.242-4.243a1 1 0 0 0 0-1.414z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
      <button
        aria-disabled="false"
        aria-label="next"
        class="emotion-16"
        tabindex="0"
      >
        <div
          class="emotion-15"
        >
          <svg
            aria-label="Chevron Right Icon"
            class=""
            fill="none"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.364 14.364a1 1 0 0 0 1.414 0l4.95-4.95.707-.707a1 1 0 0 0 0-1.414l-.707-.707-4.95-4.95a1 1 0 0 0-1.414 0l-.707.707a1 1 0 0 0 0 1.414L8.899 8l-4.242 4.243a1 1 0 0 0 0 1.414z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
